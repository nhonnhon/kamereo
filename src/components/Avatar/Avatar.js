import React from 'react'
import './Avatar.scss'

const Avatar = ({ src }) => {
  return (
    <div className={`avatar ${src ? 'has-image' : ''}`}>
      <div className="bg-image" style={{ backgroundImage: `url(${src})` }}></div>
    </div>
  )
}

export default Avatar
