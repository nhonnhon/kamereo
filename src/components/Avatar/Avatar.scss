@import '../../assets/css/base.scss';

.avatar {
  width: 100%;
  height: 300px;
  background-color: $gray;
  .bg-image {
    background-size: cover;
    background-position: center;
    height: 100%;
  }
}
