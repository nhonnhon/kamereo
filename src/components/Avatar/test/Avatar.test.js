import React from 'react'
import { create } from 'react-test-renderer'
import Avatar from '../Avatar'

describe('Avatar component', () => {
  test('Matches the snapshot', () => {
    const avatar = create(<Avatar color="white" size="big" />)
    expect(avatar.toJSON()).toMatchSnapshot()
  })
})
