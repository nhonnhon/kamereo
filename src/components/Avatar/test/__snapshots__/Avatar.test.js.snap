// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component Matches the snapshot 1`] = `
<div
  className="avatar "
>
  <div
    className="bg-image"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
</div>
`;

exports[`Button component Matches the snapshot 1`] = `
<div
  className="avatar "
>
  <div
    className="bg-image"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
</div>
`;
